### DISCUSSION TITLE
Who has a modern iOS device that can test for iOS issues?

### DISCUSSION THREAD

**Discussion by ericblade** [ericblade/quagga2#315] [General]
One of the things that we routinely saw a ton of on the other repo was people having issues with iOS.  Unfortunately, the only iOS device I have is an iPad mini 2, which will not receive a version of iOS that provides getUserMedia . . . so I am completely useless on helping to track down any issues, or write any workarounds, or really provide any help at all with iOS.

**Comment by TomasHubelbauer** [comment]
Hey Eric, I have an iPhone 10 that I can test things on. Just mention me in an issue with repro steps any time you need something and I'll get back to you when I can.

**Comment by DanielTurner** [comment]
@ericblade 
I know it doesn't give you all the functionality but to test other parts if you haven't already you could look at:
https://www.browserstack.com/open-source

**Comment by ericblade** [comment]
fwiw, my fam now has access to a modern iPad, and I can probably use that to test my app that includes quagga.  it's not something i have ability to do hardcore dev work with though.

**Comment by ericblade** [comment]
hey @TomasHubelbauer if you're still out here, or anyone else who might be interested -- someone just filed a report against the original repo, that iOS is not able to read strings larger than 12 chars from a decode?  Anyone able to give that a test?

**Comment by TomasHubelbauer** [comment]
@ericblade I'm still around and able to test. Can you give me steps on how to test?

**Comment by ericblade** [comment]
![barcode (1)](https://user-images.githubusercontent.com/1451847/96759545-32a36500-13a6-11eb-8eac-e9b587d84ffe.gif)

my guess would be to scan this (it's Code-39) and see if all 20 digits comes out .. ? the original bug report isn't exactly full of details.   https://github.com/serratus/quaggaJS/issues/441  .. I am assuming you have some kind of app that is using quagga of some version .. if not, i guess try serratus's demo page, or .. i'm hoping to be able to get my app publicly available soon :-)

**Comment by TomasHubelbauer** [comment]
![image](https://user-images.githubusercontent.com/6831144/96773456-d27fe380-13e4-11eb-8c27-dc300dc69d90.png)

Took a steady hand and a bit of time... But it did scan eventually. :-) Also I had to go to landscape mode, but it might still scan in portrait mode, it probably just requires getting the alignment just right. I tried generating Code 39 barcodes with increasing number of digits using https://barcode.tec-it.com/en/Code39FullASCII and it started struggling after 12 digit numbers. counter-intuitively, increasing the camera resolution did not make much of (if any) difference on these long codes.

**Comment by ericblade** [comment]
Thanks for checking!

So, I'm not sure exactly what you're saying -- it looks like from the screenshots it is working, but it has issues after 12 digits?  Or are you saying the generator has issues after 12 digits?

**Comment by TomasHubelbauer** [comment]
The scanner is working up until 18 digits and was able to scan the picture in this thread. Using the online Code 39 generator, after 12 digits you have to focus on aligning the camera just right more and more. Up until 12 digits you pretty much just the to get the code into frame. With more, you really want to get the code horizontal, camera in landscape orientation and get the code as wide as the frame excepting some margin so that the left most and right most code bars do not touch the  edge of the frame. But the scanner is definitely working, even if it is not effortless to scan.

**Comment by ericblade** [comment]
Thank you!
